ci-operator/step-registry/cucushift/upgrade/check/capability/cucushift-upgrade-check-capability-commands.sh

```diff
@@ -0,0 +1,451 @@
+set -o pipefail
+
+trap 'FRC=$?; create_postupgrade_junit' EXIT TERM
+
+# Generate the Junit for cap check
+function create_postupgrade_junit() {
+    echo "Generating the Junit for post upgrade check on capability"
+    filename="cluster upgrade"
+    testsuite="cluster upgrade"
+    subteam="OTA"
+    if (( FRC == 0 )); then
+        cat >"${ARTIFACT_DIR}/${filename}.xml" <<EOF
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuite name="${testsuite}" failures="0" errors="0" skipped="0" tests="1" time="1">
+  <testcase name="${subteam}:Post upgrade check on capability should succeed or skip"/>
+</testsuite>
+EOF
+    else
+        cat >"${ARTIFACT_DIR}/${filename}.xml" <<EOF
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuite name="${testsuite}" failures="1" errors="0" skipped="0" tests="1" time="1">
+  <testcase name="${subteam}:Post upgrade check on capability should succeed">
+    <failure message="">Post upgrade check on capability failed</failure>
+  </testcase>
+</testsuite>
+EOF
+    fi
+}
+
+baselinecaps_from_cluster=$(oc get clusterversion version -ojson | jq -r '.spec.capabilities.baselineCapabilitySet')
+if [[ "${baselinecaps_from_cluster}" == "" ]]; then
+    echo "spec.capabilities.baselineCapabilitySet is not set, skip the check!"
+    exit 0
+fi
+echo "baselinecaps_from_cluster: ${baselinecaps_from_cluster}"
+
+# Get capabilities string based on release version and capset
+# Need update when new cap set pops up
+function get_caps_for_version_capset() {
+    local version=$1
+    local cap_index=$2
+
+    if [ -z "${version}" ] || [ -z "${cap_index}" ]; then
+        echo "ERROR: Missing required arguments"
+        return 1
+    fi
+
+    major_version=$( echo "${version}" | cut -f1 -d. )
+    minor_version=$( echo "${version}" | cut -f2 -d. )
+    if [[ "${cap_index}" == "vCurrent" ]]; then
+        cap_index="v${major_version}.${minor_version}"
+    fi
+
+    local caps_string=""
+    case ${cap_index} in
+    "None")
+    ;;
+    "v4.11")
+    caps_string="baremetal marketplace openshift-samples"   
+    ;;
+    "v4.12")
+    caps_string="baremetal marketplace openshift-samples Console Insights Storage CSISnapshot"
+    ;;
+    "v4.13")
+    caps_string="baremetal marketplace openshift-samples Console Insights Storage CSISnapshot NodeTuning"
+    ;;
+    "v4.14")
+    caps_string="baremetal marketplace openshift-samples Console Insights Storage CSISnapshot NodeTuning MachineAPI Build DeploymentConfig ImageRegistry"
+    ;;
+    "v4.15")
+    caps_string="baremetal marketplace openshift-samples Console Insights Storage CSISnapshot NodeTuning MachineAPI Build DeploymentConfig ImageRegistry OperatorLifecycleManager"
+    ;;
+    *)
+    caps_string="baremetal marketplace openshift-samples Console Insights Storage CSISnapshot NodeTuning MachineAPI Build DeploymentConfig ImageRegistry OperatorLifecycleManager"
+    ;;
+    esac
+
+    if (( minor_version >=14 && major_version == 4 )) && [[ $caps_string != *"MachineAPI"* ]]; then
```

Comment:
We already maintain a full list for each version, I think we do not need this if block.